### DISCUSSION TITLE
Preventing retries from 404 errors?

### DISCUSSION THREAD

**Discussion by viveleroi** [sindresorhus/ky#492] [Q&A] [answered]
I've created a global ky file for my app so that we can define default config options easily. I'm calling `ky.create` with default options, exporting that, and am importing from this file in my app.

I'm testing errors and have intentionally typed an invalid url which errors with a 404. I don't want the library to retry if it receives a 404 but it tries several times before throwing a final error `HTTPError: Request failed with status code 404 Not Found`

What can I do to prevent it from re-trying 404s? I tried setting the `retry` value but that doesn't seem applicable here.

```js
export default ky.create({
  retry: 1
})
```

**Comment by sindresorhus** [comment, marked as answer]
Ky does not retry on 404, so this may be the native Fetch doing the retry. Related: https://github.com/sindresorhus/ky/issues/233

**Reply by viveleroi** [reply]
Thanks, I'm using react-query too so after I posted this I started wondering if maybe it was that, not ky. I'll have to look into that too, but it's that or fetch for sure then.

**Reply by viveleroi** [reply]
For anyone finding this, it was react-query. the `retry` config option defaults to 3.
